channel/handlers.go

```diff
@@ -762,6 +799,335 @@ func (server Server) playerUseInventoryItem(conn mnet.Client, reader mpacket.Rea
+	equip.scrollLevel++
+}
+
+// removeEquipAtSlot removes the equip from the given slot (equipped negative or inventory positive).
+func removeEquipAtSlot(plr *player, slot int16) bool {
+	if slot < 0 {
+		// Equipped item; find and clear
+		for i := range plr.equip {
+			if plr.equip[i].slotID == slot {
+				// Remove equipped item
+				plr.equip[i].amount = 0
+				return true
+			}
+		}
+		return false
+	}
+
+	// Inventory equip; remove from inventory
+	for i := range plr.equip {
+		if plr.equip[i].slotID == slot {
+			if plr.equip[i].amount != 1 {
+				return false
+			}
+			plr.equip[i].amount = 0
+			return true
+		}
+	}
+	return false
+}
+
+// findUseItemBySlot returns the use item (scroll) at the given slot from USE inventory.
+func findUseItemBySlot(plr *player, slot int16) *item {
+	for i := range plr.use {
+		if plr.use[i].slotID == slot {
+			return &plr.use[i]
+		}
+	}
+	return nil
+}
+
+// findEquipBySlot returns the equip by slot (negative = equipped, positive = inventory slot).
+func findEquipBySlot(plr *player, slot int16) *item {
+	for i := range plr.equip {
+		if plr.equip[i].slotID == slot {
+			return &plr.equip[i]
+		}
+	}
+	return nil
+}
+
+// Packet format (after ticker skip handled elsewhere):
+// - slot: int16 (USE inv slot of the buff item)
+// - itemID: int32
+// - targetName: string (length-prefixed int16)
+func (server *Server) playerUseBuffItem(conn mnet.Client, reader mpacket.Reader) {
+	plr, err := server.players.getFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	slot := reader.ReadInt16()
+	itemID := reader.ReadInt32()
+
+	// Validate item in USE inventory and item id match
+	found := false
+	for i := range plr.use {
+		if plr.use[i].slotID == slot && plr.use[i].id == itemID && plr.use[i].amount > 0 {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// Hacking / invalid
+		return
+	}
+
+	targetName := reader.ReadString(reader.ReadInt16())
+
+	// Try to resolve target by name
+	if target, err := server.players.getFromName(targetName); err == nil && target != nil {
+		// If target is the same player, original code notes needing a failure packet if it's invalid
+		// We'll still proceed if allowed by your game rules.
+		if plr.mapID != target.mapID {
+			// TODO: send an appropriate failure packet (different map)
+			plr.send(packetPlayerNoChange())
+			return
+		}
+
+		// Apply effect to target
+		// TODO: integrate actual buff application (stats/effects based on itemID)
+		applyBuffItemToPlayer(target, itemID)
+
+		// Consume the buff item from the user's USE inventory
+		if _, err := plr.takeItem(itemID, slot, 1, 2); err != nil {
+			// If consumption fails, roll back by notifying client
+			plr.send(packetPlayerNoChange())
+			return
+		}
+	} else {
+		// Target not found: send failure (TODO: precise packet)
+		plr.send(packetPlayerNoChange())
+	}
+}
+
+// applyBuffItemToPlayer is a placeholder for your buff item usage on a target player.
+func applyBuffItemToPlayer(target *player, itemID int32) {
```

Comment:
This could be re-written as `func(p* player) applyBuffItem(itemID int32)` to align with the rest of the codebase.